### ISSUE TITLE
problem about bp with optimizer

### ISSUE THREAD

**Issue by liuyijiang1994** [deeplearningathome/pytorch-language-model#1]
I change the following code:
<pre><code>
torch.nn.utils.clip_grad_norm_(model.parameters(), gradient_clip)
for p in model.parameters():
    p.data.add_(-learning_rate, p.grad.data)
total_loss += loss.item()
</pre></code>
to:
<pre><code>
torch.nn.utils.clip_grad_norm_(model.parameters(), gradient_clip)
optimizer.step()
total_loss += loss.item()
</pre></code>
where
<pre><code>
optimizer = optim.Adam(model.parameters(), lr=learning_rate, weight_decay=1e-2)
</pre></code>
and then the loss don't decreased as before.
I guss that the  function <torch.nn.utils.clip_grad_norm_>  doesn' t work for the optimizer,can anyone tell me if you know something about my problem ,much thanks!

**Comment by liuyijiang1994** [comment]
I think I find the reason,
I choose the weight decay too high to make the model convergence.
thinks
